loss-detection-app/pipelines/object_detection/1/pipeline.json

```diff
@@ -1,14 +1,12 @@
 {
-	"name": "object_detection",
-	"version": 1,
 	"type": "GStreamer",
-	"template":"urisourcebin uri=\"{source[uri]}\" ! queue ! decodebin ! videoconvert name=\"videoconvert\" ! video/x-raw ! videocrop name=\"videocrop\" ! queue leaky=upstream ! gvadetect device=CPU inference-id=inf0 model=\"{models[object_detection][1][network]}\" model-proc=\"{models[object_detection][1][proc]}\" name=\"detection\" ! gvametaconvert converter=json method=detection source=\"{source[uri]}\" name=\"jsonmetaconvert\" include-no-detections=true ! queue ! gvametapublish method=\"{destination[type]}\" address=\"{destination[host]}\" topic=\"{destination[topic]}\" timeout=1000 ! queue ! appsink name=appsink",
+	"template":"urisourcebin uri=\"{source[uri]}\" ! queue ! decodebin ! videoconvert name=\"videoconvert\" ! video/x-raw ! videocrop name=\"videocrop\" ! queue leaky=upstream ! gvadetect device=CPU model=\"{models[object_detection][1][network]}\" model-proc=\"{models[object_detection][1][proc]}\" name=\"detection\" ! gvametaconvert name=\"metaconvert\" source=\"{source[uri]}\" add-empty-results=true ! queue ! gvametapublish name=\"destination\" ! queue ! appsink name=appsink",
```

Comment:
Two suggestions: 
1) Use an array of lines (newly supported) to make things clearer. (Note lines are combined as is, be careful on spaces)

2) Make detect-device a parameter
```javascript
"template":["urisourcebin uri=\"{source[uri]}\" ! queue ! decodebin",
	    " ! videoconvert name=\"videoconvert\" ! video/x-raw ! videocrop name=\"videocrop\"",
	    " ! queue leaky=upstream",
	    " ! gvadetect device=\"{parameters[inference_device]}\" model=\"{models[object_detection][1][network]}\"",
	    " model-proc=\"{models[object_detection][1][proc]}\" name=\"detection\"",
	    " ! gvametaconvert name=\"metaconvert\" source=\"{source[uri]}\" add-empty-results=true",
	    " ! queue",
	    " ! gvametapublish name=\"destination\" ! queue ! appsink name=appsink"
	   ]
```

Reply:
Done